kernel/kernel-api/src/main/java/io/delta/kernel/internal/table/ResolvedTableBuilderImpl.java

```diff
@@ -106,7 +111,41 @@ public ResolvedTableInternal build(Engine engine) {
 
   private void validateInputOnBuild() {
     checkArgument(ctx.versionOpt.orElse(0L) >= 0, "version must be >= 0");
-    // TODO: logData only ratified staged commits
-    // TODO: logData sorted and contiguous
+    validateProtocolAndMetadataOnlyIfVersionProvided();
+    validateProtocolRead();
+    validateLogDatasOnlyRatifiedCommits();
+    validateLogDatasSortedContiguous();
```

Comment:
unsure if it's worth the effort but could we encode some of these invariants into the type system? perhaps we could make bad states unrepresentable? (probably not worth targeting immediately but maybe a follow-up if it seems a plausible improvement)

Reply:
So this is the input from the user. I think our _internal_ type system should correctly represent this -- i.e. we have `LogSegment::getAllFiles` and that is guaranteed to return a contiguous list of files, but I think having a type system for the input is overkill -- it's just a `List`, and we can really quickly and easily verify if it is contiguous or not